### ISSUE TITLE
New rule proposal: new-line-after-foreign-import

### ISSUE THREAD

**Issue by ndabAP** [eslint/eslint#12461]
**Please describe what the rule should do:**
Enforce a new line after foreign and relative (own) imports to improve the readability (distinguish easier between own imports and others).

**What category of rule is this? (place an "X" next to just one item)**

- [ ] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [x] Enforces code style (layout)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Invalid:

```js
import pkg from 'pkg'
import local from './local'
```

Valid:

```js
import pkg from 'pkg'

import local from './local'
```

**Why should this rule be included in ESLint (instead of a plugin)?**
I'm not sure if this can be achieved by a plugin.

**Are you willing to submit a pull request to implement this rule?**
No.

**Comment by mdjermanovic** [comment]
Hi @ndabAP, thanks for the rule proposal!

It looks to me that this rule could have conflicts with the `padding-line-between-statements` rule. Also (in some way) with `sort-imports`, which enforces the order of import declarations by syntax, not by the source.

Did you check the [eslint-plugin-import/order](https://github.com/benmosher/eslint-plugin-import/blob/HEAD/docs/rules/order.md) rule? It has a `newlines-between` option.

**Comment by ndabAP** [comment]
@mdjermanovic thanks for the detailed explanation. I think the `groups` option of the mentioned plugin provides exactly what I need.

**Comment by platinumazure** [comment]
Thanks @ndabAP, glad you found what you need.

I'm going to go ahead and close this issue, but please feel free to comment or create a new issue if you think ESLint core still needs an enhancement here.
